### PR TITLE
New data access better cache

### PR DESCRIPTION
**Pull request by tcoenen**
* Use cache factory
* Remove duplicated data access functions
* update datareader to support authentication headers

Solves OP-188

### CODE REVIEW

File: frontend/src/main.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ let vueApp = new Vue({
       setBuurten: 'setBuurten'
     }),
     async init () {
+      // following line implicitly initalizes the neighborhood id mappings in ./services/datasets.js
```
Comment (Vegter) [inline frontend/src/main.js:37]: Why not explicitly initialize?

File: frontend/src/services/datasets.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    // to translate between these neighborhood ids we must hit the endpoint /gastransitie/api/buurt/ .
+    getUrl: () => PRIVATE_URL + '/buurt/',
+    getCacheKey: () => 'BOOTSTRAP',
+    postProcess: results => {
+      // Initialize the mappings, then just return the results (untouched) for use in application
+      // start-up (see main.js in root of this Vue project).
+      results = asGeojson(results)
+      results.features.forEach(item => {
+        VOLLCODE2LANDELIJK[item.properties.vollcode] = item.id
+        LANDELIJK2VOLLCODE[item.id] = item.properties.vollcode
```
Comment (Vegter) [inline frontend/src/services/datasets.js:41]: VOLLCODE2LANDELIJK = results.features.reduce((obj, item) => ({ ...obj, [item.properties.vollcode]: item.id}), {})
Kan ook, maar laat maar zo

**Review by Vegter — COMMENTED** [review]
Veel beter georganiseerd, 1 vraag en 1 opmerking die je kan vergeten :-)

File: frontend/src/services/datasets.js (hunk 2 above)
Comment (tcoenen) [reply frontend/src/services/datasets.js:41]: for readability i will retain the code as is

File: frontend/src/main.js (hunk 1 above)
Comment (tcoenen) [reply frontend/src/main.js:37]: ./services/datasets.js needs to initalize before being used. I chose to make it impossible to not run module initialization (and in the process made initialization implicit). I hooked this code up to the app initialization to make sure the app is in a known state, before anything else can happen.

**Pull request merged**